The job **`i686-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11451629834/job/31861175918) [(plain)](https://github.com/rust-lang-ci/rust/commit/8b863d7d722c8655bc8604ed986d15353f3d5428/checks/31861175918/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [assembly] tests/assembly/x86-return-float.rs#normal stdout ----

error in revision `normal`: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/i686-unknown-linux-gnu/llvm/build/bin/FileCheck" "--input-file" "/checkout/obj/build/i686-unknown-linux-gnu/test/assembly/x86-return-float.normal/x86-return-float.s" "/checkout/tests/assembly/x86-return-float.rs" "--check-prefix=CHECK" "--check-prefix" "normal" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100" "--implicit-check-not" "fld" "--implicit-check-not" "fst"
--- stderr -------------------------------
/checkout/tests/assembly/x86-return-float.rs:308:12: error: CHECK: expected string not found in input
/checkout/tests/assembly/x86-return-float.rs:308:12: error: CHECK: expected string not found in input
 // CHECK: pinsrw $0, {{.*}}(%ebp), %xmm0
           ^
/checkout/obj/build/i686-unknown-linux-gnu/test/assembly/x86-return-float.normal/x86-return-float.s:621:12: note: scanning from here
return_f16:
           ^
/checkout/obj/build/i686-unknown-linux-gnu/test/assembly/x86-return-float.normal/x86-return-float.s:628:6: note: possible intended match here
 movzwl 8(%ebp), %eax


Input file: /checkout/obj/build/i686-unknown-linux-gnu/test/assembly/x86-return-float.normal/x86-return-float.s


-dump-input=help explains the following input dump.
Input was:
<<<<<<
             .
             .
             .
             .
           521:  calll get_f64_other@PLT 
           522:  subl $4, %esp 
           523:  movsd -24(%ebp), %xmm0 
           524:  movl -16(%ebp), %eax 
           525:  movsd %xmm0, (%esi) 
           526:  movl %eax, 8(%esi) 
           527:  addl $32, %esp 
           528:  popl %esi 
           529:  popl %ebx 
           530:  popl %ebp 
           531:  .cfi_def_cfa %esp, 4 
           532:  retl 
           533: .Lfunc_end17: 
           534:  .size call_f64_other, .Lfunc_end17-call_f64_other 
           535:  .cfi_endproc 
           536:  
           537:  .section .text.call_other_f32,"ax",@progbits 
           538:  .globl call_other_f32 
           539:  .p2align 4, 0x90 
           540:  .type call_other_f32,@function 
           541: call_other_f32: 
           542:  .cfi_startproc 
           543:  pushl %ebp 
           544:  .cfi_def_cfa_offset 8 
           545:  .cfi_offset %ebp, -8 
           546:  movl %esp, %ebp 
           547:  .cfi_def_cfa_register %ebp 
           548:  pushl %ebx 
           549:  pushl %esi 
           550:  subl $16, %esp 
           551:  .cfi_offset %esi, -16 
           552:  .cfi_offset %ebx, -12 
           553:  movl 8(%ebp), %esi 
           554:  calll .L18$pb 
           555: .L18$pb: 
           556:  popl %ebx 
           557:  leal -16(%ebp), %eax 
           558: .Ltmp8: 
           559:  addl $_GLOBAL_OFFSET_TABLE_+(.Ltmp8-.L18$pb), %ebx 
           560:  movl %eax, (%esp) 
           561:  calll get_other_f32@PLT 
           562:  subl $4, %esp 
           563:  movl -16(%ebp), %eax 
           564:  movss -12(%ebp), %xmm0 
           565:  movl %eax, (%esi) 
           566:  movss %xmm0, 4(%esi) 
           567:  addl $16, %esp 
           568:  popl %esi 
           569:  popl %ebx 
           570:  popl %ebp 
           571:  .cfi_def_cfa %esp, 4 
           572:  retl 
           573: .Lfunc_end18: 
           574:  .size call_other_f32, .Lfunc_end18-call_other_f32 
           575:  .cfi_endproc 
           576:  
           577:  .section .text.call_other_f64,"ax",@progbits 
           578:  .globl call_other_f64 
           579:  .p2align 4, 0x90 
           580:  .type call_other_f64,@function 
           581: call_other_f64: 
           582:  .cfi_startproc 
           583:  pushl %ebp 
           584:  .cfi_def_cfa_offset 8 
           585:  .cfi_offset %ebp, -8 
           586:  movl %esp, %ebp 
           587:  .cfi_def_cfa_register %ebp 
           588:  pushl %ebx 
           589:  pushl %esi 
           590:  subl $16, %esp 
           591:  .cfi_offset %esi, -16 
           592:  .cfi_offset %ebx, -12 
           593:  movl 8(%ebp), %esi 
           594:  calll .L19$pb 
           595: .L19$pb: 
           596:  popl %ebx 
           597:  leal -20(%ebp), %eax 
           598: .Ltmp9: 
           599:  addl $_GLOBAL_OFFSET_TABLE_+(.Ltmp9-.L19$pb), %ebx 
           600:  movl %eax, (%esp) 
           601:  calll get_other_f64@PLT 
           602:  subl $4, %esp 
           603:  movl -20(%ebp), %eax 
           604:  movsd -16(%ebp), %xmm0 
           605:  movl %eax, (%esi) 
           606:  movsd %xmm0, 4(%esi) 
           607:  addl $16, %esp 
           608:  popl %esi 
           609:  popl %ebx 
           610:  popl %ebp 
           611:  .cfi_def_cfa %esp, 4 
           612:  retl 
           613: .Lfunc_end19: 
           614:  .size call_other_f64, .Lfunc_end19-call_other_f64 
           615:  .cfi_endproc 
           616:  
           617:  .section .text.return_f16,"ax",@progbits 
           618:  .globl return_f16 
           619:  .p2align 4, 0x90 
           620:  .type return_f16,@function 
           621: return_f16: 
check:308'0                X error: no match found
           622:  .cfi_startproc 
check:308'0     ~~~~~~~~~~~~~~~~
           623:  pushl %ebp 
check:308'0     ~~~~~~~~~~~~
           624:  .cfi_def_cfa_offset 8 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~~
           625:  .cfi_offset %ebp, -8 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~
           626:  movl %esp, %ebp 
check:308'0     ~~~~~~~~~~~~~~~~~
           627:  .cfi_def_cfa_register %ebp 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           628:  movzwl 8(%ebp), %eax 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~
check:308'1          ?                 possible intended match
           629:  popl %ebp 
check:308'0     ~~~~~~~~~~~
           630:  .cfi_def_cfa %esp, 4 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~
           631:  retl 
check:308'0     ~~~~~~
           632: .Lfunc_end20: 
check:308'0     ~~~~~~~~~~~~~~
           633:  .size return_f16, .Lfunc_end20-return_f16 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           634:  .cfi_endproc 
check:308'0     ~~~~~~~~~~~~~~
check:308'0     ~
check:308'0     ~
           636:  .section .text.return_f128,"ax",@progbits 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           637:  .globl return_f128 
check:308'0     ~~~~~~~~~~~~~~~~~~~~
           638:  .p2align 4, 0x90 
check:308'0     ~~~~~~~~~~~~~~~~~~
           639:  .type return_f128,@function 
check:308'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           640: return_f128: 
check:308'0     ~~~~~~~~~~~~
           641:  .cfi_startproc 
           642:  pushl %ebp 
           643:  .cfi_def_cfa_offset 8 
           644:  .cfi_offset %ebp, -8 
           645:  movl %esp, %ebp 
           646:  .cfi_def_cfa_register %ebp 
           647:  pushl %edi 
           648:  pushl %esi 
           649:  .cfi_offset %esi, -16 
           650:  .cfi_offset %edi, -12 
           651:  movl 8(%ebp), %eax 
           652:  movl 12(%ebp), %ecx 
           653:  movl 16(%ebp), %edx 
           654:  movl 20(%ebp), %esi 
           655:  movl 24(%ebp), %edi 
           656:  movl %edi, 12(%eax) 
           657:  movl %esi, 8(%eax) 
           658:  movl %edx, 4(%eax) 
           659:  movl %ecx, (%eax) 
           660:  popl %esi 
           661:  popl %edi 
           662:  popl %ebp 
           663:  .cfi_def_cfa %esp, 4 
           664:  retl $4 
           665: .Lfunc_end21: 
           666:  .size return_f128, .Lfunc_end21-return_f128 
           667:  .cfi_endproc 
           668:  
           669:  .ident "rustc version 1.84.0-nightly (8b863d7d7 2024-10-22)" 
           670:  .section ".note.GNU-stack","",@progbits 
------------------------------------------



```

</details>

--- a/tests/assembly/x86-return-float.rs
+++ b/tests/assembly/x86-return-float.rs
@@ -305,8 +305,10 @@ pub unsafe fn call_other_f64(x: &mut (usize, f64)) {
 // CHECK-LABEL: return_f16:
 #[no_mangle]
 pub fn return_f16(x: f16) -> f16 {
-    // CHECK: pinsrw $0, {{.*}}(%ebp), %xmm0
-    // CHECK-NOT: xmm0
+    // CHECK: pushl %ebp
+    // CHECK: movl %esp, %ebp
+    // CHECK: movzwl 8(%ebp), %eax
+    // CHECK: popl %ebp
     // CHECK: retl
     x
 }

